grace/io/image_dataset.py

```diff
@@ -70,9 +70,7 @@ def __getitem__(self, idx: int) -> tuple[torch.Tensor, dict]:
         img_path = self.image_paths[idx]
         grace_path = self.grace_paths[idx]
 
-        image = torch.tensor(
-            self.image_reader_fn(img_path), dtype=torch.float32
-        )
+        image = torch.tensor(self.image_reader_fn(img_path).astype("float32"))
```

Comment:
This works, but shouldn't the data types & loading be dealt with in the `reader` functions? In this way, we're getting the image post-processed here but potentially not in the rest of the pipeline.

Reply:
I thought the idea was that the conversion wouldn't be done automatically in the rest of the pipeline? Such that we can preserve the original data types. I.e., only where it's needed (where it's out to be fed to a model) do we convert to `float32`

Reply:
Ohhh now I see what you meant! I think it makes perfect sense, but should we make sure that other datatypes get process correctly here before merging in? (I'm probably just overly cautious) 👀